Code/GraphMol/MarvinParse/MarvinDefs.cpp

```diff
@@ -0,0 +1,2978 @@
+//
+//  Copyright (C) 2002-2022 Greg Landrum and other RDKit contributors
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include <RDGeneral/RDLog.h>
+#include "MarvinDefs.h"
+
+namespace RDKit {
+
+thread_local std::map<std::string, std::string> sgMap;
+thread_local std::map<std::string, std::string> atomMap;
+thread_local std::map<std::string, std::string> bondMap;
```

Comment:
These cannot be part of MarvinMolBase  since they keep the mappings from old to new for ALL the MarvinMolBase derived entitities at the same time.  I did go ahead and make that parameters passed into the routines that need them, but this really does not look cleaner.

Reply:
In my experience, thread_locals are a really heavy hammer and has some potential annoying side effects with linkage.

It seems that these data structures could only exist for the lifespan of the parsers and be sent into parseMolecule.